src/ComputationContainer/Share/Compare.hpp

```diff
@@ -109,7 +110,140 @@ bool operator<(const Share<FixedPoint> &left, const FixedPoint &right);
 bool operator==(const Share<FixedPoint> &left, const FixedPoint &right);
 
 Share<FixedPoint> LTZ(const Share<FixedPoint> &s);
+/// @brief l=32 split array
+inline constexpr int delta[] = {5, 5, 5, 5, 5, 5, 2};
+/// @brief generate one-hot-vector used by unitv
+/// @tparam N
+/// @return random share [r] and r-th value is 1 ,otherwise is 0 in n length array
+template <typename T, std::enable_if_t<std::is_integral_v<T>, std::nullptr_t> = nullptr>
+Share<T> operator==(const Share<T> &left, const Share<T> &right)
+{
+    return equality(left, right);
+}
+template <size_t N>
+std::pair<Share<int>, std::vector<Share<int>>> unitvPrep()
+{
+    std::vector<Share<int>> e(N * N);
+    std::vector<AddressId> addressIds(N * N);
+    std::vector<Share<int>> ret(N * N);
+    Config *conf = Config::getInstance();
+    auto server = ComputationToComputation::Server::getServer();
+    auto random_s = RandGenerator::getInstance()->getRandVec<long long>(1, 1 << 20, N);
+    auto r = RandGenerator::getInstance()->getRand<long long>(0, 31);
```

Comment:
is `r` related to `N`?
```suggestion
    auto r = RandGenerator::getInstance()->getRand<long long>(0, N - 1);
```

Reply:
I checked it, thank you.